05-classification.Rmd

```diff
@@ -663,10 +688,19 @@ scaled_cancer <- bake(uc_recipe, unscaled_cancer)
 scaled_cancer
 ```
 
-Now let's generate the two scatter plots, one for `unscaled_cancer` and one for
-`scaled_cancer`, and show them side-by-side. Each has the same new observation
-annotated with its $K=3$ nearest neighbours:
-
+Figure \@ref(fig:05-scaling-plt) shows the two scatter plots side-by-side&mdash;one for `unscaled_cancer` and one for
+`scaled_cancer`. Each has the same new observation annotated with its $K=3$ nearest neighbours.
+In the plot for the nonstandardized original data, you can see some odd choices
+for the three nearest neighbours. In particular, the "neighbours" are visually
+well within the cloud of benign observations, and the neighbours are all nearly
+vertically aligned with the new observation (which is why it looks like there
+is only one black line on this plot).  Here the computation of nearest
+neighbours is dominated by the much larger-scale area variable. On the right,
+the plot for standardized data shows a much more intuitively reasonable
+selection of nearest neighbours. Thus, standardizing the data can change things
+in an important way when we are using predictive algorithms.  As a rule of
+thumb, standardizing your data should be a part of the preprocessing you do
+before any predictive modelling / analysis.
 
 ```{r 05-scaling-plt, echo = FALSE, fig.height = 4, fig.width = 10, fig.cap = "Comparison of K = 3 nearest neighbours with standardized and unstandardized data"}
```

Comment:
I find the plot on the left hard to interpret. Maybe an inset where we zoom in could help? The trouble is you can't actually see the neighbours...

Reply:
Great idea, but I don't have time to do this. I'm going to open a new issue for it.